### PR TITLE
EWB-4105: Update super pom to 0.34.1

### PR DESCRIPTION
**Pull request by Lax125**
# Description

Use super pom version 0.34.1. This comes with breaking changes for Vert.x.

# Associated tasks

Tasks blocking this one:
- [x] https://github.com/zepben/evolve-super-pom/pull/36
- [x] https://github.com/zepben/test-utils-jvm/pull/4
- [x] https://github.com/zepben/vertx-utils/pull/3

Tasks this is blocking:
- https://github.com/zepben/evolve-sdk-jvm/pull/153
- https://github.com/zepben/ewb-network-routes/pull/127
- https://github.com/zepben/energy-workbench-server/pull/129
- https://github.com/zepben/ewb-network-server/pull/59
- https://github.com/zepben/noose/pull/4

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [x] I have performed a self review of my own code (including checking issues raised when creating the PR).
- [x] I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.
- [ ] ~I have commented my code in any hard-to-understand or hacky areas.~
- [x] I have handled all new warnings generated by the compiler or IDE.
- [x] I have rebased onto the target branch (usually main).
      
### Documentation
- [x] I have updated the changelog.
- [ ] ~I have updated any documentation required for these changes.~

# Breaking Changes
- [x] I have considered if this is a breaking change and will communicate it with other team members if so.

Updates Vert.x major version (3 -> 4). Removed unused `vertx` module.

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [EWB-4105 Update evolve conn jvm](https://app.clickup.com/t/6929263/EWB-4105)

**Comment by Lax125** [comment]
Will fix up some sections using deprecated functions

**Comment by Lax125** [comment]
Looks like auth config route is still needed...

File: src/main/kotlin/com/zepben/auth/server/AuthConfigRoute.kt [hunk 1]
```diff
@@ -29,7 +21,7 @@ import io.vertx.ext.web.RoutingContext
 
 private data class AuthConfigResponse(
     val authType: AuthMethod,
-    val issuerDomain: String,
+    val issuer: String,
```
Comment (Lax125) [inline src/main/kotlin/com/zepben/auth/server/AuthConfigRoute.kt:24]: `ewb-web-client` currently uses the `issuer` field in the auth config json.

File: src/test/kotlin/com/zepben/auth/server/AuthConfigRouteTest.kt
Comment (charlta) [inline src/test/kotlin/com/zepben/auth/server/AuthConfigRouteTest.kt:1]: todo: This file should not have been removed

**Review by charlta — CHANGES_REQUESTED** [review]

File: src/test/kotlin/com/zepben/auth/server/AuthConfigRouteTest.kt
Comment (Lax125) [reply src/test/kotlin/com/zepben/auth/server/AuthConfigRouteTest.kt:1]: Added back with updated copyright and test. Also edited changelog to mention auth config field name change.

**Review by charlta — APPROVED** [review]

**Pull request merged**